Source/Core/Helpers.cs

```diff
@@ -263,18 +264,39 @@ public static void ExtraTraceInformation(CoreOptions options, string point)
       }
     }
 
+    private static readonly HashSet<string> UsedLogNames = new();
```

Comment:
I think you can simplify the code below a lot by using a concurrent dictionary from file names to indices here: this way you can replace the while loop and the lock with a single call to `AddOrUpdate`.